src/index.ts

```diff
@@ -0,0 +1,103 @@
+import * as comments from './comments/index.js';
+import { handleComments, printComment } from './slang-comments/index.js';
+import massageAstNode from './clean.js';
+import loc from './loc.js';
+import options from './options.js';
+import solidityParse from './parser.js';
+import solidityPrint from './printer.js';
+import parse from './slangParser.js';
+import print from './slangPrinter.js';
+import { isComment, isBlockComment } from './slang-utils/is-comment.js';
+import { locEnd, locStart } from './slang-utils/loc.js';
+
+import type {
+  Parser,
+  Printer,
+  RequiredOptions,
+  SupportLanguage
+} from 'prettier';
+import type { AstNode } from './slang-nodes';
+
+const parserName = 'slang';
+const astFormat = 'slang-ast';
+
+// https://prettier.io/docs/en/plugins.html#languages
```

Comment:
is `slang-ast` the public user-facing "parser" name that uses are going to specify in their config?
In that case, what do you think about `slang-solidity` instead?

1. "ast" is a feature name, not tied to the parser, and might indicate otherwise.
2. allows supporting a `slang-yul` parser in the future, if it was supported: https://docs.soliditylang.org/en/latest/yul.html#stand-alone-usage